xdsl/interactive/app.py

```diff
@@ -97,19 +97,24 @@ def get_all_possible_rewrites(
     return res
 
 
-def condensed_pass_list(input: builtin.ModuleOp) -> tuple[type[ModulePass], ...]:
+def get_condensed_pass_list(
+    input: builtin.ModuleOp,
+) -> tuple[tuple[str, type[ModulePass], PipelinePassSpec | None], ...]:
     """Returns a tuple of passes (pass name and pass instance) that modify the IR."""
 
     ctx = MLContext(True)
 
     for dialect_name, dialect_factory in get_all_dialects().items():
         ctx.register_dialect(dialect_name, dialect_factory)
 
-    selections: list[type[ModulePass]] = []
+    selections: tuple[tuple[str, type[ModulePass], PipelinePassSpec | None], ...] = ()
```

Comment:
adding to a list in a for loop is much more efficient than appending to a tuple, could you factor this back, please?